Update publishing warnings

H3ll3m4 opened the pull request:
## Context
Publishing warnings are being updated. Some of the content has to be changed but also the display of autocomplete warnings with the other warnings.

## Implementation
In order to change the content it is a simple update of the translation file en.yml. In order to have the publishing warnings for autocomplete uniform with the rest, we add the publish them in the list if presenters for the publishing page presenter and remove them from the publish environment view.

## Tests
Unit tests and publishing acceptance tests have been updated. Local testing with manual observation performed.

njseeto commented on app/presenters/autocomplete_items_presenter.rb:
```diff
@@ -23,15 +26,22 @@ def pages_with_autocomplete_component
   end
 
   def messages
```
```suggestion
  def message
```

Are we able to rename this method to `message` and then delete the new `message` method?

njseeto commented on app/presenters/autocomplete_items_presenter.rb:
```diff
@@ -23,15 +26,22 @@ def pages_with_autocomplete_component
+      page.components.each do |component|
+        next unless component.uuid.in?(component_uuids_without_items)
+
+        link = govuk_link_to(component.humanised_title, Rails.application.routes.url_helpers.edit_page_path(service.service_id, page.uuid))
```
```suggestion

```
could we move this method to a private method? Then we can just call it in the `description` variable. Something like:

```
  def messages
    pages_with_autocomplete_component.each_with_object([]) do |page, arry|
      page.components.each do |component|
        next unless component.uuid.in?(component_uuids_without_items)

        description = I18n.t(
          "publish.autocomplete_items.#{deployment_environment}.message",
          title: link(component, page)
        )

        msg = description.html_safe
        arry.push(msg)
      end
    end
  end

  private

  def link(component, page)
    govuk_link_to(
      component.humanised_title,
      Rails.application.routes.url_helpers.edit_page_path(service.service_id, page.uuid)
    )
  end
```

njseeto commented on app/presenters/autocomplete_items_presenter.rb:
```diff
@@ -23,15 +26,22 @@ def pages_with_autocomplete_component
+
+        link = govuk_link_to(component.humanised_title, Rails.application.routes.url_helpers.edit_page_path(service.service_id, page.uuid))
+
+        description = I18n.t("publish.autocomplete_items.#{deployment_environment}.message", title: link)
```
```suggestion
        msg = I18n.t(
          "publish.autocomplete_items.#{deployment_environment}.message",
          title: link
        ).html_safe
```
Not a biggie, but we could chain the `html_safe` in the variable assignment and rename the variable, then we can delete the `msg = description.html_safe` line.

njseeto commented on spec/presenters/publishing_page_presenter_spec.rb:
```diff
@@ -34,13 +34,17 @@
     context 'submission warning presenters' do
       let(:presenters) do
-        [submission_pages_presenter, from_address_presenter]
+        [submission_pages_presenter,
+         autocomplete_warning,
+         from_address_presenter]
```
```suggestion
        [
          submission_pages_presenter,
          autocomplete_warning,
          from_address_presenter
        ]
```

Small formatting suggestion

njseeto reviewed (commented):
Looks great, just a couple of suggestions.

brenetic commented on app/presenters/publishing_page_presenter.rb:
```diff
@@ -59,6 +59,10 @@ def submission_pages_presenter
-    [submission_pages_presenter, from_address_presenter]
+    presenters = [submission_pages_presenter, from_address_presenter]
+
+    return presenters if deployment_environment == 'dev'
```
why do we not want the autocomplete presenter to be there if it's dev?

brenetic commented:
can we have a screenshot please?

i may have completely misunderstood the story, but are the autocomplete warnings now part of the list of _all_ publishing warnings on the publishing page? if that's the case then maintaining the `autocomplete_warning` partial might not be necessary as the `SubmissionWarningsPresenter` will just map over any presenters that it's initialized with and spit out the messages

H3ll3m4 commented:
We moved the autocomplete warning in the first header group only for publishing to Live. However, as it has been clarified this morning after the stand up, we want the publishing to Test to have autocomplete warnings separated, so like before we started the work on this feature. The results looks like this, with a server running locally : 
<img width="756" alt="Screenshot 2022-11-10 at 16 44 08" src="https://user-images.githubusercontent.com/26165112/201156032-11778d57-2bf3-42d9-9cd4-6e024ae20bd0.png">

In addition, we added a few updates of the content, following version 1.3 of the [publishing warnings](https://docs.google.com/spreadsheets/d/11FvkKJcO1Yfn3bVOM9IPSD0_LHjqgTSJmASx33lYe-E/edit#gid=814462722), see tab 1.3, and only the text written ToDo on column K.

brenetic commented on app/views/publish/_autocomplete_warning.html.erb:
```diff
@@ -1,8 +1,6 @@
-<% autocomplete_warning.messages.each do |message| %>
+<% autocomplete_warning.message.each do |message| %>
```
is there only one message now per presenter? if so do we need to loop? if not we could keep both methods

brenetic commented on app/presenters/autocomplete_items_presenter.rb:
```diff
@@ -23,15 +26,24 @@ def pages_with_autocomplete_component
+  # the method message is required by the submission warning presenter
+  def message
+    messages
+  end
```
`alias :message, :messages`

brenetic reviewed (approved):
🚀
